### ISSUE TITLE
logger from shared.ts eats error messages

### ISSUE THREAD

**Issue by fungusakafungus** [aws-samples/cloudfront-authorization-at-edge#164]
try e.g. something like this:

```
let logger = new Logger(LogLevel["debug"]);
try {
  JSON.parse("{")
} catch(error) {
  logger.error(error)
  console.error(error)
}
```
it will print "{}" for logger from [shared.ts](https://github.com/aws-samples/cloudfront-authorization-at-edge/blob/master/src/lambda-edge/shared/shared.ts#L107), and "Unexpected end of JSON input" for console.error.

See full example [here](https://www.typescriptlang.org/play?ssl=44&ssc=1&pln=50&pc=2#code/KYOwrgtgBAMg9gcxsAbsANlA3gKClAIhDhGAKgF4oAGAGj0OACcm4nyqBGOhggdwCGTEBygAmHvgIBLEADM4ogMyTCAE2AAjMAlEAWHgF8cAY3QCAzhdiIEzbAxMkLAFyZgTLtgAoADk2kUARdgKHREZDR0AC4bJFQMAEpsYwZ-QODQgCsLEmk5AE9vIQQLWIEQAoBtAF1k3Hx8JmAXMGEoEosAOggBX29ipgRyyuSKAD4HRsb8qG8XAt9gODkOocoKKgI4TSzgTwJ6hmn8NwKpk+nm1vaAKQBlAHkAOS7XAJAEfKKSxIBuY4nQxQEzBEwACwulyaLTaIDWCAB0KgxkuqJO1zhCKRjUM-wY6N8YE06GkJigsgU3i6NM6IwKR2ms3m4Ok3XC8SiUHGVHgnIwXUpcEZJycIFy6GAXQ51JpLlZ3RyeUKg1KiXx03RhOJpPJgmEsq6dI6oyhFNWLLZ0oiCUwPLikQF+pAIumYolUudhvlVqVIG+qos6px+C1aR1ZKgzFYTENxoqDLNzJ97JtXPtfMd6C60bYrsa7rgkpzLB8NK6Ka6foDnWDgLD+CJJMjGm0CDjQzKJsTDRmFsrHKz3N5aYFrZ0+fwheLbgEJmA3oVVdy-pVtY1uIJOGMkpcYVs9iopD4cTsscztqqBHHujqALOFweLy6viEFgXBCwh23ILB4O8uZMCKHJniWMYAaWQGOM4RZSoBEExok25AA)

**Comment by ottokruse** [comment]
Mmm that is an unfortunate and silly bug :|

Thanks for reporting!

Probably should just skip the JSON stringify-ing, and just log, keep it simple.

Interested in taking that and PR-ing?

**Comment by fungusakafungus** [comment]
I'm not really comfortable with typescript, it costed me some hours to understand what was happening... Also, I've seen some log lines including `... some_key: Object ...` after trying with console.log, so there might be usefulness in stringify

**Comment by ottokruse** [comment]
Yeah the stringify-ing was to avoid logging any non-useful `[object Object]`.
Let's figure out something nicer.

**Comment by ottokruse** [comment]
Created a PR with a fix: #165 . Let me know what you think
